File: WorkIt/app/src/main/java/prot3ct/workit/views/edit_profile/EditProfileActivity.java
package prot3ct.workit.views.edit_profile;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import prot3ct.workit.R;
import prot3ct.workit.views.edit_profile.base.EditProfileContract;

public class EditProfileActivity extends AppCompatActivity {
    private EditProfileContract.Presenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_profile);

        EditProfileFragment editProfileFragment =
                (EditProfileFragment) getSupportFragmentManager().findFragmentById(R.id.fragment_container);

        if (editProfileFragment == null) {
            editProfileFragment = EditProfileFragment.newInstance();

            getSupportFragmentManager()
                    .beginTransaction()
                    .add(R.id.fragment_container, editProfileFragment)
                    .commit();
        }

        this.presenter = new EditProfilePresenter(editProfileFragment, this);
        editProfileFragment.setPresenter(this.presenter);
    }
}

File: WorkIt/app/src/main/java/prot3ct/workit/views/edit_profile/EditProfileActivity.kt
package prot3ct.workit.views.edit_profile

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import prot3ct.workit.R
import prot3ct.workit.views.edit_profile.EditProfileFragment
import prot3ct.workit.views.edit_profile.EditProfilePresenter
import prot3ct.workit.views.edit_profile.base.EditProfileContract

class EditProfileActivity : AppCompatActivity() {

    private lateinit var presenter: EditProfileContract.Presenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_profile)
        var editProfileFragment =
            supportFragmentManager.findFragmentById(R.id.fragment_container) as EditProfileFragment?
        if (editProfileFragment == null) {
            editProfileFragment = EditProfileFragment.newInstance()
            supportFragmentManager
                .beginTransaction()
                .add(R.id.fragment_container, editProfileFragment)
                .commit()
        }
        presenter = EditProfilePresenter(editProfileFragment, this)
        editProfileFragment.setPresenter(presenter)
    }
}